### PR TITLE
feat: add pin property

### PR DESCRIPTION
**Pull request by lunsdorf**
> Implements issues #4, although I named the property pin instead of fixed since everything related is named like that.

This PR applies all changes from #131 as a patch together with an additional fix for the unit tests. I did it this way, because I deleted to fork I worked on years ago and couldn't figure out how to append to an existing PR with all sources deleted.

I did this because people keep asking if the changes could be integrated. I did not re-evaluate or actually test the changes I made when I originally wrote this. So somebody should definitly do that.

### CODE REVIEW

**Comment by janczizikow** [comment]
Hey @lunsdorf thanks for the PR! Looks like a nice thing to add! I'll try to have a closer look at it this week. Meanwhile I had a question that came to me while glancing at it initially:

What if the `pin` property changes once the Headroom has been already rendered? I think there might be some additional logic that should be added either in `componentDidUpdate` or `getDerivedStateFromProps`.

Also would be great if you could mention the prop in the docs (`www/pages/index.md`)

**Comment by lunsdorf** [comment]
Ok, I added the prop in the docs and also check for `pin` property changes in `componentDidUpdate`. I'm reusing the scroll handler function for this, as this is the one that uses `requestAnimationFrame` callback. The naming is a bit unfortunate however.

File: src/index.js [hunk 1]
```diff
@@ -148,6 +150,8 @@ export default class Headroom extends Component {
             this.eventListenerOptions
           )
       }
+    } else if (prevProps.pin !== this.props.pin) {
+      this.handleScroll()
```
Comment (janczizikow) [inline src/index.js:154]: Is there any reason to put it in `else if` instead of `if`? Previous conditions check if props changed from `disabled: true` to `disabled: false` or the other way around. Those two cases are handled differently but it's unrelated to change of `pin` prop. So I think you could change it to a separate `if` instead of `else if`.

**Review by janczizikow — APPROVED** [review]
LGTM 🎉
